src/mgr.sol

```diff
@@ -186,128 +205,131 @@ contract TinlakeManager {
+        }
+        else if (what == "rwaToken") {
+            rwaToken = GemLike(data);
+        }
+        else revert("TinlakeMgr/file-unknown-param");
     }
 
     // --- Liquidation ---
-    function tell() public auth {
-        require(safe, "TinlakeManager/not-safe");
-        (uint256 ink, ) = vat.urns(ilk, address(this));
+    // triggers a soft liquidation of the DROP collateral
+    // a redeemOrder is submitted to receive DAI back
+    function tell() public governance {
+        require(safe, "TinlakeMgr/not-safe");
+        uint256 ink = gem.balanceOf(address(this));
         safe = false;
         gem.approve(tranche, ink);
         pool.redeemOrder(ink);
         emit Tell(ink);
     }
 
+    // triggers the payout of a DROP redemption
+    // method can be called multiple times after the liquidation until all
+    // DROP tokens are redeemed
     function unwind(uint256 endEpoch) public {
-        require(!safe && glad && live, "TinlakeManager/not-soft-liquidation");
-        (uint256 redeemed, , ,uint256 remainingDrop) = pool.disburse(endEpoch);
-        (uint256 ink, uint256 art) = vat.urns(ilk, address(this));
-        uint256 dropReturned = sub(ink, remainingDrop);
-        require(dropReturned <= 2 ** 255, "TinlakeManager/overflow");
+        require(!safe && glad && live, "TinlakeMgr/not-soft-liquidation");
+
+        (uint256 redeemed, , ,) = pool.disburse(endEpoch);
 
+        // here we use the urn instead of address(this)
+        (, uint256 art) = vat.urns(ilk, address(urn));
         (, uint256 rate, , ,) = vat.ilks(ilk);
+
         uint256 cdptab = mul(art, rate);
+
         uint256 payBack = min(redeemed, divup(cdptab, RAY));
 
-        daiJoin.join(address(this), payBack);
-        // Repay dai debt up to the full amount
-        // and exit the gems used up
-        uint256 dart = mul(RAY, payBack) / rate;
-        require(dart <= 2 ** 255, "TinlakeManager/overflow");
-        vat.frob(ilk, address(this), address(this), address(this),
-                 0, -int256(dart));
-        vat.grab(ilk, address(this), address(this), address(this),
-                 -int256(dropReturned), 0);
-        vat.slip(ilk, address(this), -int256(dropReturned));
+        dai.transferFrom(address(this), address(urn), payBack);
+        urn.wipe(payBack);
         // Return possible remainder to the owner
-        dai.transfer(operator, dai.balanceOf(address(this)));
+        dai.transfer(owner, dai.balanceOf(address(this)));
         emit Unwind(payBack);
     }
 
-    // --- Writeoff ---
-    function sink() public auth {
-        require(!safe && glad && live, "TinlakeManager/bad-state");
-        (uint256 ink, uint256 art) = vat.urns(ilk, address(this));
-        require(ink <= 2 ** 255, "TinlakeManager/overflow");
-        require(art <= 2 ** 255, "TinlakeManager/overflow");
+    // --- Write-off ---
+    // method should be called before RwaLiquidationOracle.cull()
+    function sink() public governance {
```

Comment:
I don't think there's a way to get the tab from the `RwaLiquidationOracle`. Am I missing something?

Reply:
This would be the minimal change we need I believe to make it work:

```
contract Liq is RwaLiquidationOracle {
    uint public tab;
    constructor (address vat_, address vow_) RwaLiquidationOracle(vat_, vow_) public{}
    function cull(bytes32 ilk, address urn) external auth {
        (uint256 ink, uint256 art) = vat.urns(ilk, urn);
        tab = mul(rate, art);
        super.cull(ilk, urn);
    }
}
```